components/encryption/src/manager/mod.rs

```diff
@@ -586,17 +603,30 @@ impl DataKeyManager {
 
 impl Drop for DataKeyManager {
     fn drop(&mut self) {
-        self.rotate_terminal.send(()).unwrap();
-        self.background_worker.take().unwrap().join().unwrap();
+        self.rotate_terminal
+            .send(())
+            .expect("DataKeyManager drop send");
+        self.background_worker
+            .take()
+            .expect("DataKeyManager worker take")
+            .join()
+            .expect("DataKeyManager worker join");
     }
 }
 
 impl EncryptionKeyManager for DataKeyManager {
     // Get key to open existing file.
     fn get_file(&self, fname: &str) -> IoResult<FileEncryptionInfo> {
         let (method, key_id, iv) = {
-            let file = self.dicts.get_file(fname);
-            (file.method, file.key_id, file.iv)
+            match self.dicts.get_file(fname) {
+                Some(file) => (file.method, file.key_id, file.iv),
+                None => {
```

Comment:
then why we need to refactor `get_file`? `get_file_exists` looks redundant. Again if the file entry doesn't exist in file dictionary, it could mean the file does not exist, or the file exists but not encrypted. In the key manager we don't care about if the file exists on the file system. We return file info with method = plaintext and let the caller decide what to do.

Reply:
Or in another word, we always assume the caller of `get_file` knows the file exists.